core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/hybrid/HybridRoleManager.java

```diff
@@ -632,6 +714,66 @@ private String truncateInternalDomainFromFilter(String filter) {
         return hybridRoleListOfUsers;
     }
 
+    /**
+     * Get hybrid role list of groups.
+     *
+     * @param groupNames group name list.
+     * @return map of hybrid role list of groups.
+     * @throws UserStoreException userStoreException.
+     */
+    public Map<String, List<String>> getHybridRoleListOfGroups(List<String> groupNames, String domainName)
+            throws UserStoreException {
+
+        Map<String, List<String>> hybridRoleListOfGroups = new HashMap<>();
+        String sqlStmt = realmConfig.getRealmProperty(HybridJDBCConstants.GET_ROLE_LIST_OF_GROUPS);
+        StringBuilder groupNameParameter = new StringBuilder();
+
+        if (StringUtils.isEmpty(sqlStmt)) {
+            sqlStmt = HybridJDBCConstants.GET_INTERNAL_ROLE_LIST_OF_GROUPS_SQL;
+        }
+        for (int i = 0; i < groupNames.size(); i++) {
+
+            groupNames.set(i, groupNames.get(i).replaceAll("'", "''"));
+            groupNameParameter.append("'").append(groupNames.get(i)).append("'");
+
+            if (i != groupNames.size() - 1) {
+                groupNameParameter.append(",");
+            }
+        }
+
+        sqlStmt = sqlStmt.replaceFirst("\\?", Matcher.quoteReplacement(groupNameParameter.toString()));
```

Comment:
why can't we do this using prepStmt?

Reply:
prepStmt is not supported for comma-separated strings. So we will have to do it this way for IN (,).